test/engine/NamedResultCacheTest.cpp

```diff
@@ -115,7 +119,7 @@ TEST(NamedResultCache, E2E) {
   std::string pinnedQuery =
       "SELECT * { {?s <p> <o> } UNION {VALUES ?s { <notInVocab> }}} INTERNAL "
       "SORT BY ?s";
-  qec->pinResultWithName() = "dummyQuery";
+  qec->pinResultWithName() = {"dummyQuery"};
```

Comment:
Maybe add a test that also includes a  geoIndex?  Depending on how easy/complicated this is.